### PR TITLE
IL-249: Create e2e test for FavoriteList

### PR DESCRIPTION
**Pull request by andrew13pol**
add:

- 2 json files for sending a request and checking the response
- Class with test methods

### CODE REVIEW

File: src/test/java/com/zufar/icedlatte/favorite/api/endpoint/FavoriteListEndpointTest.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    protected String password;
+    protected static String jwtToken = "";
+
+    private static final String AUTHENTICATE_TEMPLATE = "/security/model/authenticate-template.json";
+
+    private static final String FAVORITE_LIST_SCHEMA = "favorite/model/schema/favorite-list-schema.json";
+    private static final String PRODUCT_ADD_TO_FAVORITE_LIST = "/favorite/model/product-add-to-favorite-list.json";
+
+    @DynamicPropertySource
+    static void dataSourceProperties(DynamicPropertyRegistry registry) {
```
Comment (yevr19) [inline src/test/java/com/zufar/icedlatte/favorite/api/endpoint/FavoriteListEndpointTest.java:63]: I dont think we need to use @DynamicPropertySource in cohesion  with @ServiceConnection

File: src/test/java/com/zufar/icedlatte/favorite/api/endpoint/FavoriteListEndpointTest.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    void shouldAddProductToFavoriteList() {
+        String body = getRequestBody(PRODUCT_ADD_TO_FAVORITE_LIST);
+
+        Response response = given(specification)
+                .body(body)
+                .post();
+
+        assertRestApiOkResponse(response, FAVORITE_LIST_SCHEMA);
+    }
+
```
Comment (yevr19) [inline src/test/java/com/zufar/icedlatte/favorite/api/endpoint/FavoriteListEndpointTest.java:119]: we also need more tests on negative scenarios

File: src/test/java/com/zufar/icedlatte/favorite/api/endpoint/FavoriteListEndpointTest.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    @Value("${jwt.expiration}")
+    protected Long expiration;
+
+    @Value("${jwt.email}")
+    protected String email;
+
+    @Value("${jwt.password}")
+    protected String password;
+    protected static String jwtToken = "";
```
Comment (yevr19) [inline src/test/java/com/zufar/icedlatte/favorite/api/endpoint/FavoriteListEndpointTest.java:55]: consider to use this variable locally and return it from getJwtToken method and then reuse it in tokenAndSpecification()

File: src/test/java/com/zufar/icedlatte/favorite/api/endpoint/FavoriteListEndpointTest.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+    @Test
+    @DisplayName("Should delete product from FavoriteList")
+    void shouldDeleteProductFromFavoriteList() {
+        String productId = "418499f3-d951-40bf-9414-5cb90ab21ecb";
+
+        Response response = given(specification)
+                .delete("/{productId}", productId);
+
+        assertRestApiEmptyBodyResponse(response, HttpStatus.valueOf(200));
```
Comment (yevr19) [inline src/test/java/com/zufar/icedlatte/favorite/api/endpoint/FavoriteListEndpointTest.java:128]: consider to use HTTPStatus.OK constant instead of plain 200 value

File: src/test/java/com/zufar/icedlatte/favorite/api/endpoint/FavoriteListEndpointTest.java [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+                .accept(ContentType.JSON);
+
+        String body = getRequestBody(AUTHENTICATE_TEMPLATE)
+                .formatted(email, password);
+
+        Response response = given(specification)
+                .body(body)
+                .post("/authenticate");
+
+        jwtToken = response.getBody().path("token");
```
Comment (yevr19) [inline src/test/java/com/zufar/icedlatte/favorite/api/endpoint/FavoriteListEndpointTest.java:105]: should we validate the return value forn null or empty to avoid inconsistency ?

File: src/test/java/com/zufar/icedlatte/favorite/api/endpoint/FavoriteListEndpointTest.java [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+                .formatted(email, password);
+
+        Response response = given(specification)
+                .body(body)
+                .post("/authenticate");
+
+        String jwtToken = response.getBody().path("token");
+
+        if (isJwtTokenNotValid(jwtToken)) {
+            throw new RuntimeException("JWT Token is empty or null. Test failed.");
```
Comment (yevr19) [inline src/test/java/com/zufar/icedlatte/favorite/api/endpoint/FavoriteListEndpointTest.java:105]: be more consice and choose not a generic exception. At least you can throw IllegalArgumentException instead

**Review by yevr19 — APPROVED** [review]

**Review by Sunagatov — APPROVED** [review]
Approved
